docs/building-extensions/plugins/plugin-events/application.md

```diff
@@ -0,0 +1,381 @@
+sidebar_position: 1
+title: Application / System Events
+toc_max_heading_level: 2 
+---
+
+Application / System Events
+===========================
+
+Application / System Events are triggered by the core Joomla application code - 
+the highest level code which provides the main flow of control in Joomla.
+
+For a plugin to capture these events it should be defined as a system plugin. 
+
+In this category we also include the events onBeforeExtensionBoot and onAfterExtensionBoot,
+as they don't fit into any other category, and your plugin has to be a system plugin to receive them all. 
+
+:::warning
+  System plugins are loaded every time Joomla runs, on both the site front-end and administrator back-end. 
+If you have a PHP syntax error in a system plugin then you will get locked out of the Joomla administrator back-end, 
+and will have to go into phpmyadmin (or equivalent) to set to 0 the `enabled` field within the plugin record in the `#__extensions` table. 
+:::
+
+## Joomla overview
+
+The overview below of how Joomla handles an HTTP request on the front-end of a website
+is designed to give you an idea of how the system events fit into the overall control flow.
+If you would like further information then you may find it helpful to watch the video 
+[How Joomla Works - a guide for extension developers](https://youtu.be/JKnq47Yhtvs).
+
+- Initialisation 
+- trigger *onAfterInitialise*
+- Route - to work out the main component, and associated parameters (view, layout, etc)
+- trigger *onAfterRoute*
+- Instantiate the Document object representing the output - normally HtmlDocument
+- trigger *onAfterInitialiseDocument*
+- run the component - the output from the component is stored in a buffer
+- trigger *onAfterDispatch*
+- reads and parses the template file (eg templates/cassiopeia/index.php) which forms the basis of what is returned to the client
+- trigger *onBeforeRender*
+- processes the template file, filling in the `<jdoc:include type="..." />` "holes" there, for the various types:
+  - type="metas" - trigger *onBeforeCompileHead*, and produce metadata to include in the `<head>`
+  - type="styles", type="scripts" - include CSS and Javascript
+  - type="component" - include the component output
+  - type="message" - include entries for the system message area
+  - type="modules" - run the code of each of the modules to be displayed on that page, and include the output
+- trigger *onAfterRender*
+- if gzip compression is required, then compress the output and trigger *onAfterCompress*
+- trigger *onBeforeRespond*
+- send the HTTP response with the HTML
+- trigger *onAfterRespond*
+- tidy up, including storing the session data
+
+## Application Events overview
+
+The application event classes can be found in libraries/src/Event/Application. 
+In general they have a single parameter `subject` which is usually set to `$this` in the code which dispatches the event.
+There is no getter method for `subject` so you have to use something like:
```

Comment:
All Application events have `getApplication()` method, which provided by base `ApplicationEvent` class.
So Developers can access it as `$event->getApplication()`